lib/bank_accounts/checking_account.rb

```diff
@@ -0,0 +1,100 @@
+module BankAccounts
+  class CheckingAccount < Account
+    ##--------------------------------------------------------------------------
+    # CONSTANTS
+
+
+    # limits
+    CHECKS_LIMIT = 3
+    MINIMUM_BALANCE = -10
+
+    # fees
+    EXCESS_CHECKS_FEE = 2.00
+    WITHDRAWAL_FEE = 1.00
+
+
+    ##--------------------------------------------------------------------------
+    # INSTANCE METHODS
+    attr_reader :checks_used
+
+
+    def initialize(id, initial_balance)
+      super(id, initial_balance)
+
+      @checks_used = 0
+
+      return true
+    end
+
+
+    def reset_checks
+      @checks_used = 0
+
+      return true
+    end
+
+
+    # withdraw money from account.
+    def withdraw(amount)
+      if super(amount + WITHDRAWAL_FEE)
+        return @balance
+      else
+        return @balance
+      end
+    end
```

Comment:
See (this comment)[https://github.com/Ada-Developers-Academy/C3Projects--BankAccounts/pull/6/files#diff-5671ee19f228f2b2c21bd3392819282bR56] (sorry again!)